File: numeron-annotation/src/main/java/com/erzbir/numeron/annotation/DefaultScope.java
package com.erzbir.numeron.annotation;

import kotlin.coroutines.CoroutineContext;
import kotlin.coroutines.EmptyCoroutineContext;
import kotlinx.coroutines.CoroutineScope;
import org.jetbrains.annotations.NotNull;

/**
 * @author Erzbir
 * @Data 2023/7/24
 */
public class DefaultScope implements CoroutineScope {

    @NotNull
    @Override
    public CoroutineContext getCoroutineContext() {
        return EmptyCoroutineContext.INSTANCE;
    }
}

File: numeron-annotation/src/main/java/com/erzbir/numeron/annotation/DefaultScope.kt
package com.erzbir.numeron.annotation

import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.Job
import kotlin.coroutines.CoroutineContext

/**
 * @author Erzbir
 * @Data 2023/7/26
 */
class DefaultScope : CoroutineScope {
    private val job = Job()


    override val coroutineContext: CoroutineContext = Dispatchers.Default + job

    fun cancelJob() {
        job.cancel()
    }

    companion object {
        @JvmField
        val INSTANCE = DefaultScope()
    }
}